assets/js/settings/blocks/store-api-nonce.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import { getSetting } from '../shared';
+
+/**
+ * Returns whether or not this is a non GET wc/store API request.
+ *
+ * @param {Object} options Options given to apiFetch.
+ *
+ * @return boolean
+ */
+const isStoreApiGetRequest = ( options ) => {
+	const url = options.url || options.path;
+	if ( ! url || ! options.method || options.method === 'GET' ) {
+		return false;
+	}
+	return /wc\/store\//.exec( url ) !== null;
+};
+
+/** cache for nonce initialized from hydration  */
+let nonce = getSetting( 'storeApiNonce' );
+
+const setNonce = ( headers ) => {
+	const newNonce = headers?.get( 'X-WC-Store-API-Nonce' );
+	if ( newNonce ) {
+		nonce = newNonce;
+	}
+};
```

Comment:
This is a public api on `apiFetch.setNonce` so we should probably doc block this (Note: I realize there's lots of code I've merged into master that's a public api without doc blocks during this cycle that still needs it, so if you want to do as a follow-up I understand and support).